lib/index.js

```diff
@@ -717,7 +740,13 @@ if (!enabled) {
   if (exports.cfg.hostnameAlias) {
     options.hostnameAlias = exports.cfg.hostnameAlias
   }
-  bindings.oboeInit(serviceKey, options)
+
+  // delete the environment variable, init oboe, and restore it. this is
+  // done because oboe will prefer the environment variable to anything
+  // specified here.
+  delete env.APPOPTICS_SERVICE_KEY
+  bindings.oboeInit(cleansedKey, options)
+  env.APPOPTICS_SERVICE_KEY = environmentKey
```

Comment:
Is there any imaginable scenario, where a service with a different config.serviceKey would be started from this env? If so, the env var should only be set if that is where the service_name was read from in the first place.